### PR TITLE
Offer unpacked code from package.json

### PR DESCRIPTION
**Pull request by cwillisf**
### Proposed Changes

This change declares `./src/index.js` as the "main" script in `package.json`, and stops building `dist/node/*` with webpack. This change also restores the ES6 version of the test scripts, recently altered for Node.js 4.x compatibility.

### Reason for Changes

This sets us up to build `scratch-gui` in a single step, allowing webpack to better optimize our files and reducing the duplication of code within our output bundles.

### Test Coverage

Test coverage has not changed -- just the syntax of the test scripts.

### CODE REVIEW

**Comment by cwillisf** [comment]
Tests are broken due to the `require("binary-loader!...")` change. Will fix tomorrow...

File: webpack.config.js [hunk 1]
```diff
@@ -1,21 +1,26 @@
-const path = require('path');
 const webpack = require('webpack');
 
-const base = {
+module.exports = {
+    devtool: 'cheap-module-source-map',
```
Comment (cwillisf) [inline webpack.config.js:4]: This option has treated me well over the past few days, though I'm happy to discuss and experiment with other options.
See also: https://webpack.js.org/configuration/devtool/#devtool

File: webpack.config.js [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
+        'scratch-storage.min': './src/index-web.js'
+    },
     module: {
         rules: [
             {
-                include: [
-                    path.resolve(__dirname, 'src')
-                ],
-                test: /\.js$/,
+                test: /[\\/]+scratch-[^\\/]+[\\/]+src[\\/]+.+\.js$/,
```
Comment (rschamp) [inline webpack.config.js:12]: Same comment as on the GUI — I think it would be more understandable if this was more explicit/verbose.

**Comment by cwillisf** [comment]
Tests are working now, through the magic of jsdom.
Team: please give 2859875 some extra scrutiny to make sure you can stomach doing things this way :)

**Comment by cwillisf** [comment]
The test diffs are more readable if you ignore white-space changes by adding `?w=1` to the end of the URL, by the way:
https://github.com/LLK/scratch-storage/pull/6/files?w=1

**Comment by rschamp** [comment]
Whoops my comment didn't go through: set to needs work for the babel loader regex.

**Comment by cwillisf** [comment]
Closing this in favor of #8 for now. There are a few other changes, like proper version numbers, which should happen before I try this kind of change again.

**Pull request closed without merging**